src/procedures/proc_betweenness.c

```
@@ -0,0 +1,614 @@
+	// allocate sources array
+	GrB_Index *sources = rm_malloc(sizeof(GrB_Index) * (*samplingSize));
+
+	// seed the random number generator
+    srand(samplingSeed);
+
+	// pick random sources
+	for(int i = 0; i < *samplingSize; i++) {
+		uint64_t x;
+		GrB_Index idx = rand() % container->nvals;
+
+		info = GrB_Vector_extractElement(&x, container->i, idx);
```

Cppcheck (2.10-2)

[error] Memory leak

(memleak)